lib/config/rules/typescript.js

```diff
@@ -123,4 +119,66 @@ module.exports = {
   '@typescript-eslint/restrict-plus-operands': 'error',
+  // Bans “// @ts-ignore” comments from being used
+  '@typescript-eslint/ban-ts-ignore': 'error',
+  // Enforce consistent brace style for blocks
+  '@typescript-eslint/brace-style': 'error',
+  // Enforces consistent usage of type assertions.
+  '@typescript-eslint/consistent-type-assertions': 'error',
+  // Consistent with type definition either interface or type
+  '@typescript-eslint/consistent-type-definitions': 'error',
+  // Require or disallow spacing between function identifiers and their invocations
+  '@typescript-eslint/func-call-spacing': 'error',
+  // Disallow empty functions
+  '@typescript-eslint/no-empty-function': 'error',
+  // Disallow unnecessary parentheses
+  '@typescript-eslint/no-extra-parens': 'error',
+  // Requires Promise-like values to be handled appropriately.
+  '@typescript-eslint/no-floating-promises': 'off',
+  // Disallows magic numbers
+  '@typescript-eslint/no-magic-numbers': 'off',
+  // Prevents conditionals where the type is always truthy or always falsy
+  '@typescript-eslint/no-unnecessary-condition': 'error',
+  // Warns if an explicitly specified type argument is the default for that type parameter
+  '@typescript-eslint/no-unnecessary-type-arguments': 'error',
+  // Prefer a ‘for-of’ loop over a standard ‘for’ loop if the index is only used to access the array being iterated
+  '@typescript-eslint/prefer-for-of': 'error',
+  // Requires that private members are marked as readonly if they're never modified outside of the constructor
+  '@typescript-eslint/prefer-readonly': 'error',
+  // Enforce the consistent use of either backticks, double, or single quotes
+  '@typescript-eslint/quotes': [
+    'error',
+    'single',
+    {avoidEscape: true, allowTemplateLiterals: true},
+  ],
+  // Enforce giving compare argument to Array#sort
+  '@typescript-eslint/require-array-sort-compare': 'error',
+  // Require or disallow semicolons instead of ASI
+  '@typescript-eslint/semi': 'error',
+  // Restricts the types allowed in boolean expressions
+  '@typescript-eslint/strict-boolean-expressions': 'off',
+  // Sets preference level for triple slash directives versus ES6-style import declarations
+  '@typescript-eslint/triple-slash-reference': 'error',
+  // Requires type annotations to exist
+  '@typescript-eslint/typedef': 'off',
+  // Warns for any two overloads that could be unified into one by using a union or an optional/rest parameter
+  '@typescript-eslint/unified-signatures': 'error',
+  // Disallows awaiting a value that is not a Thenable
+  '@typescript-eslint/await-thenable': 'error',
+  // Disallow iterating over an array with a for-in loop
+  '@typescript-eslint/no-for-in-array': 'error',
+  // Avoid using promises in places not designed to handle them
+  '@typescript-eslint/no-misused-promises': 'error',
+  // Warns if a type assertion does not change the type of an expression
+  '@typescript-eslint/no-unnecessary-type-assertion': 'error',
```

Comment:
some people will hate this rule

Reply:
Why do you say that?

Reply:
I think there's an argument to be made that `const foo: Foo = createFoo()` is valuable even if `createFoo` always returns `Foo` and the explicit type annotation is therefore unnecessary. Maybe less so for `const foo: string = 'foo';` but in general there are enough times that the explicitness is valuable that I think it's worth leaving it to dev discretion.